apps/rewards/app/app-state-reducer.js

```diff
@@ -20,16 +21,17 @@ function appStateReducer(state) {
 
   if(state){
     state.amountTokens = state.balances.map(token => { 
-      return {amount: token.amount, symbol: token.symbol, address: token.address, decimals: token.decimals}
+      return { amount: token.amount, symbol: token.symbol, address: token.address, decimals: token.decimals }
     })
     state.rewards = state.rewards  || []
     state.claims = state.claims  || []
     state.rewards = state.rewards.map(reward => {
       if(reward.isMerit){
         reward.rewardType = ONE_TIME_MERIT
         reward.dateReference = new Date()
-      } else if (reward.occurances === 1){
+      } else if (reward.occurances.toString() === '1'){
         reward.rewardType = ONE_TIME_DIVIDEND
+        reward.dateReference = new Date(reward.endDate)
```

Comment:
Did you check this and verify that `reward.endDate` will be in the correct format? Do we need to multiply by 1000?

Reply:
This is correct, no need to multiply it by 1000.